topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -464,15 +486,99 @@ private void updateInternal(List<ReconcilableTopic> topics) {
         // execute those updates
         var alterConfigsResults = alterConfigs(someAlterConfigs);
         var createPartitionsResults = createPartitions(someCreatePartitions);
-
-        accumulateResults(results, currentStatesOrError, alterConfigsResults, createPartitionsResults);
-
+        var findDifferentRfResults = findDifferentRf(currentStatesOrError);
+        
+        // Cruise Control integration
+        maybeResetReplicasChangeStatuses(currentStatesOrError);
+        maybeCheckReplicasChanges(topics, results, findDifferentRfResults);
```

Comment:
I really don't find how we're passing around this mutable `results` to be very intuitive. Is there no way to refactor these methods to return some new results, which maybe we merge into the results here, rather than passing `results` deep down the call stack?

Reply:
I agree, but this is something that was already there in other methods, so I'll open a dedicated PR once this is merged.